File: LeetCode/problems/594-longest-harmonious-subsequence/594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int ans = 0;
        for(int L = 0; L < nums.length; L++){
            int R = Arrays.binarySearch(nums,nums[L]+1);
            if(R>0){
                while(R<nums.length && nums[R] < nums[L]+2){
                    R++;
                }
                ans = Math.max(ans, R-L);
            }
        }
        return ans;
    }
}

File: LeetCode/problems/594-longest-harmonious-subsequence/594-longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    let count = _.countBy(nums);
    let max = 0;
    for(let num of nums){
        if(count[num+1]){
            max = Math.max(max, count[num]+count[num+1]);
        }
    }
    return max;
};
